docker-compose.test.yml

```diff
@@ -0,0 +1,12 @@
+machine-test:
+  build: .
+  dockerfile: Dockerfile.test
```

Comment:
My hunch is that you could rename this file to `docker-compose.yml`, remove the `Dockerfile` line for the compose yml file, and not make the default `CMD` in the `Dockerfile` run the tests. 

Instead, have the `docker-compose.yml` file look like this:

```
machine:
  build: .
  environment:
    - DATABASE_URL=postgresql://openaddr:openaddr@postgres/openaddr
  links:
    - postgres

postgres:
  image: mdillon/postgis
  environment:
    - POSTGRES_USER=openaddr
    - POSTGRES_PASSWORD=openaddr
```

and then run tests like this:
```
docker-compose run --rm machine python setup.py test
```
or maybe just
```
docker-compose run --rm machine ./setup.py test
```
Once you're there you could add a `Makefile` that turns that into a call to `make test` :)